di-ipv-dcmaw-async-stub/lambdas/src/handlers/tokenHandler.ts

```diff
@@ -0,0 +1,44 @@
+import { APIGatewayProxyEventV2, APIGatewayProxyResultV2 } from "aws-lambda";
+import { buildApiResponse } from "../common/apiResponse";
+import auth from "basic-auth";
+import getConfig from "../common/config";
+import getErrorMessage from "../common/errorReporting";
+
+export async function handler(
+  event: APIGatewayProxyEventV2,
+): Promise<APIGatewayProxyResultV2> {
+  try {
+    const config = await getConfig();
+
+    // Check authorization
+    const credentials = auth(event);
+    if (
+      credentials === undefined ||
+      credentials.name !== config.dummyClientId ||
+      credentials.pass !== config.dummySecret
+    ) {
+      return buildApiResponse({ errorMessage: "Invalid credentials" }, 401);
+    }
+
+    if (!event.body?.includes("grant_type=client_credentials")) {
```

Comment:
not a big deal but this check should probably be before the auth check?

Reply:
It can, but I don't understand why it would be better?
In a real system you probably don't want to tell people without good credentials what else they've got wrong - but obviously this isn't a real system.

Reply:
Fair point, fine to leave